libPhoneNumberGeocoding/NBPhoneNumberOfflineGeocoder.m

```diff
@@ -0,0 +1,150 @@
+//
+//  NBPhoneNumberOfflineGeocoder.m
+//  libPhoneNumberiOS
+//
+//  Created by Rastaar Haghi on 6/12/20.
+//  Copyright © 2020 Google LLC. All rights reserved.
+//
+
+#import "NBPhoneNumberOfflineGeocoder.h"
+#import "Metadata/NBGeocoderMetadataHelper.h"
+#import "NBPhoneNumber.h"
+#import "NBPhoneNumberUtil.h"
+
+@implementation NBPhoneNumberOfflineGeocoder {
+ @private
+  NBPhoneNumberUtil *_phoneNumberUtil;
+  NSCache<NSString *, NBGeocoderMetadataHelper *> *_metadataHelpers;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self != nil) {
+    _phoneNumberUtil = NBPhoneNumberUtil.sharedInstance;
+    _metadataHelpers = [[NSCache alloc] init];
+  }
+  return self;
+}
+
+- (nullable NSString *)countryNameForNumber:(NBPhoneNumber *)number
+                           withLanguageCode:(NSString *)languageCode {
+  NSArray *regionCodes = [_phoneNumberUtil getRegionCodesForCountryCode:number.countryCode];
+  if ([regionCodes count] == 1) {
+    return [self regionDisplayName:regionCodes[0] withLanguageCode:languageCode];
+  } else {
+    NSString *regionWhereNumberIsValid = @"ZZ";
```

Comment:
Can you extract the `@"ZZ"` since it's used multiple times.

Reply:
Done!